packages/connection-storage/src/connection-storage.spec.ts

```diff
@@ -27,36 +28,41 @@ function getConnectionInfo(props: Partial<ConnectionInfo> = {}) {
   };
 }
 
-function writeFakeConnection(
+async function writeFakeConnection(
   tmpDir: string,
   connection: { connectionInfo: ConnectionInfo }
 ) {
   const filePath = getConnectionFilePath(tmpDir, connection.connectionInfo.id);
   const connectionsDir = path.dirname(filePath);
-  fs.mkdirSync(connectionsDir, { recursive: true });
-  fs.writeFileSync(filePath, JSON.stringify(connection));
+  await fs.mkdir(connectionsDir, { recursive: true });
+  await fs.writeFile(filePath, JSON.stringify(connection));
```

Comment:
Here we use `JSON` but in `UserData` we have EJSON, this is a great test to make sure we can read from the previous format, maybe not what we want for all tests?